### PR TITLE
Add next-redux-saga types

### PR DESCRIPTION
**Pull request by leocavalcante**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

Closes https://github.com/bmealhouse/next-redux-saga/issues/5

### CODE REVIEW

File: types/next-redux-saga/tslint.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
```
Comment (plantain-00) [inline types/next-redux-saga/tslint.json]: For new packages, this file should only `"extends": "dtslint/dt.json"`

**Review by plantain-00 — CHANGES_REQUESTED** [review]

**Review by plantain-00 — APPROVED** [review]

File: types/next-redux-saga/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+// Definitions by: Leo Cavalcante <https://github.com/leocavalcante>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
+
+import { ComponentType } from "react";
+
+declare function nextReduxSaga<P = any>(arg: { async: boolean }): ((BaseComponent: ComponentType<P>) => ComponentType<P>);
+declare function nextReduxSaga<P = any>(arg: ComponentType<P>): ComponentType<P>;
+
+export default nextReduxSaga;
```
Comment (andy-ms) [inline types/next-redux-saga/index.d.ts]: `require("next-redux-saga")` is itself a function, not an object with a `default` property. So this should be `export = nextReduxSaga;`.

**Review by ghost — CHANGES_REQUESTED** [review]

File: types/next-redux-saga/next-redux-saga-tests.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import withRedux from 'next-redux-wrapper';
+import withReduxSaga from 'next-redux-saga';
```
Comment (andy-ms) [inline types/next-redux-saga/next-redux-saga-tests.tsx]: This should be `import withReduxSaga = require('next-redux-saga')`, or you could add `"esModuleInterop": true` to `tsconfig.json` compiler options.

**Comment by leocavalcante** [comment]
Awesome! cc @bmealhouse

**Comment by bmealhouse** [comment]
Thanks @leocavalcante!

**Pull request merged**